build.gradle

```diff
@@ -20,25 +20,83 @@ dependencies {
 
 // First, apply the publishing plugin
 buildscript {
-  repositories {
-    maven {
-      url "https://plugins.gradle.org/m2/"
+    repositories {
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
+        jcenter()
+    }
+    dependencies {
+        classpath "com.gradle.publish:plugin-publish-plugin:0.9.6"
+        classpath 'de.undercouch:gradle-download-task:3.1.2'
+        classpath "gradle.plugin.org.unbroken-dome.gradle-plugins:gradle-testsets-plugin:1.3.2"
     }
-	jcenter()
-  }
-  dependencies {
-    classpath "com.gradle.publish:plugin-publish-plugin:0.9.6"
-	classpath 'de.undercouch:gradle-download-task:3.1.2'
-  }
 }
 
+/* 
+  The TestSets plugin simplifies integration testing. By using the plugin, we just need to 
+  declare an integration test 'testSet' for an integrationTest task to be automatically generated for us. See 
+  https://www.petrikainulainen.net/programming/gradle/getting-started-with-gradle-integration-testing-with-the-testsets-plugin/
+*/
+apply plugin: "org.unbroken-dome.test-sets"
+
 apply plugin: 'de.undercouch.download'
 apply plugin: "com.gradle.plugin-publish"
 apply plugin: 'maven'
 
+// Declare our integration test set
+testSets {
+    integrationTest
+}
+
+/*
+  Ensure that the outputs of the integration task are never "up-to-date" so that 
+  the integration test task runs fully every time we invoke it.
+*/
+project.integrationTest {
+    outputs.upToDateWhen { false }
+}
+
+/*
+  Integration tests are run before the check task. Check is run as part of the build task, 
+  making the check test depend on integrationTest ensures integration tests are part of the build task.
+*/
+check.dependsOn integrationTest 
+// Run unit tests before integration tests
+integrationTest.mustRunAfter test
+
+task copyUmpleJar(type: Copy) {
```

Comment:
Yes, it's for testing. It copies the Umple jar from it's location in the main Umple plugin project to the TestProject directory. That way we get rid of the "long and dangly path" in TestProject's `build.gradle` file.